lib/ansible/modules/web_infrastructure/ansible_tower/tower_send.py

```diff
@@ -0,0 +1,167 @@
+      required: false
+      default: 'default'
+      choices: ["default", "random"]
+
+notes:
+  - One of assets or files needs to be passed in
+
+requirements:
+  - "ansible-tower-cli >= 3.3.0"
+  - six.moves.StringIO
+  - sys
+
+extends_documentation_fragment: tower
+'''
+
+EXAMPLES = '''
+- name: Import all tower assets
+  tower_send:
+    assets: "{{ export_output.assets }}"
+    tower_config_file: "~/tower_cli.cfg"
+'''
+
+RETURN = '''
+output:
+    description: The import messages
+    returned: success, fail
+    type: list
+    sample: [ 'Message 1', 'Messag 2' ]
+'''
+
+import os
+import sys
+
+from ansible.module_utils.six.moves import StringIO
+from ansible.module_utils.ansible_tower import tower_argument_spec, tower_auth_config, HAS_TOWER_CLI
+from ansible.module_utils.basic import AnsibleModule
+
+from tempfile import mkstemp
+
+try:
+    from tower_cli.cli.transfer.send import Sender
+    from tower_cli.utils.exceptions import TowerCLIError
+
+    from tower_cli.conf import settings
+    TOWER_CLI_HAS_EXPORT = True
+except ImportError:
+    TOWER_CLI_HAS_EXPORT = False
+
+
+def main():
+    argument_spec = tower_argument_spec()
+    argument_spec.update(dict(
+        assets=dict(required=False, default=''),
+        files=dict(required=False, default=[], type='list'),
+        prevent=dict(required=False, default=[], type='list'),
+        password_management=dict(required=False, default='default', choices=['default', 'random']),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+
+    if not HAS_TOWER_CLI:
+        module.fail_json(msg='ansible-tower-cli required for this module')
+
+    if not TOWER_CLI_HAS_EXPORT:
+        module.fail_json(msg='ansible-tower-cli version does not support export')
+
+    assets = module.params.get('assets')
+    prevent = module.params.get('prevent')
+    password_management = module.params.get('password_management')
+    files = module.params.get('files')
+
+    result = dict(
+        changed=False,
+        msg='',
+        output='',
+    )
+
+    if assets == '' and files == []:
+        result['msg'] = "Assets or files must be specified"
+        module.fail_json(**result)
+
+    path = None
+    if assets != '':
+        # We got assets so we need to dump this out to a temp file and append that to files
+        handle, path = mkstemp(prefix='', suffix='', dir='')
+        with open(path, 'w') as f:
+            f.write(assets)
+        f.close()
+        files.append(path)
+
+    old_stdout = sys.stdout
+    sys.stdout = captured_stdout = StringIO()
+
+    tower_auth = tower_auth_config(module)
+    failed = False
+    with settings.runtime_values(**tower_auth):
+        try:
+            sender = Sender(no_color=False)
+            sender.send(files, prevent, password_management)
+            if sender.error_messages > 0:
+                failed = True
+                result['msg'] = "Transfer Failed with %d errors" % sender.error_messages
```

Comment:
Should we attempt to print the error messages in some way, or is it just too verbose?

Reply:
They are written to stdout so this line is grabbing it with the rest of the output:
```
result['output'] = captured_stdout.getvalue().split("\n")
```